apisix/plugins/grpc-transcode/util.lua

```diff
@@ -51,6 +53,35 @@ function _M.find_method(proto, service, method)
 end
 
 
+function _M.set_options(proto, options)
+    local cur_opts = proto.options
+    if cur_opts then
+        if cur_opts == options then
+            return
+        end
+
+        local same = true
+        table.sort(options)
+        for i, v in ipairs(options) do
+            if cur_opts[i] ~= v then
+                same = false
+                break
+            end
+        end
```

Comment:
I assume the extra check of pointer and values are for perf improvements. But just a small heads up, this check will fail on the second time the `set_option` method is called and on the first call, the `cur_opts` was nil.

In that scenario, at the first run, the unsorted table will be cached - L81.
L66 will try to match with the sorted table `options`. 

Anyway, just a small nitpicking indeed : )

Reply:
Good catch!